pkg/model/faucet/faucet_test.go

```diff
@@ -0,0 +1,425 @@
+	env.TestEnv.AssertLedgerBalance(env.Wallet1, wallet1Balance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet2, wallet2Balance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet3, wallet3Balance)
+
+	// 29 Mi - 10 Mi = 19 Mi
+	err := env.RequestFundsAndIssueMilestone([]*utils.HDWallet{env.Wallet1})
+	require.NoError(t, err)
+
+	faucetBalance -= faucetAmount
+	env.AssertFaucetBalance(faucetBalance)
+
+	// 19 Mi - 10 Mi = 9 Mi
+	err = env.RequestFundsAndIssueMilestone([]*utils.HDWallet{env.Wallet2})
+	require.NoError(t, err)
+
+	faucetBalance -= faucetAmount
+	env.AssertFaucetBalance(faucetBalance)
+
+	// 9 Mi - 10 Mi = error
+	err = env.RequestFundsAndIssueMilestone([]*utils.HDWallet{env.Wallet3})
+	require.Error(t, err)
+
+	env.AssertFaucetBalance(faucetBalance)
+}
+
+func TestCollectFaucetFunds(t *testing.T) {
+	// check if faucet collects funds if no requests left
+
+	var faucetBalance uint64 = 1000000000         //  1 Gi
+	var wallet1Balance uint64 = 0                 //  0  i
+	var wallet2Balance uint64 = 0                 //  0  i
+	var wallet3Balance uint64 = 0                 //  0  i
+	var faucetAmount uint64 = 10000000            // 10 Mi
+	var faucetSmallAmount uint64 = 1000000        // 1  Mi
+	var faucetMaxAddressBalance uint64 = 20000000 // 20 Mi
+
+	env := test.NewFaucetTestEnv(t,
+		faucetBalance,
+		wallet1Balance,
+		wallet2Balance,
+		wallet3Balance,
+		faucetAmount,
+		faucetSmallAmount,
+		faucetMaxAddressBalance,
+		false)
+	defer env.Cleanup()
+	require.NotNil(t, env)
+
+	confirmedMilestoneIndex := env.ConfirmedMilestoneIndex() // 4
+	require.Equal(t, milestone.Index(4), confirmedMilestoneIndex)
+
+	// Verify balances
+	genesisBalance := iotago.TokenSupply - faucetBalance - wallet1Balance - wallet2Balance - wallet3Balance
+
+	env.AssertFaucetBalance(faucetBalance)
+	env.TestEnv.AssertLedgerBalance(env.GenesisWallet, genesisBalance)
+	env.TestEnv.AssertLedgerBalance(env.FaucetWallet, faucetBalance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet1, wallet1Balance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet2, wallet2Balance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet3, wallet3Balance)
+
+	env.AssertAddressUTXOCount(env.FaucetWallet.Address(), 1)
+
+	err := env.RequestFundsAndIssueMilestone([]*utils.HDWallet{env.Wallet1})
+	require.NoError(t, err)
+
+	env.AssertAddressUTXOCount(env.FaucetWallet.Address(), 1)
+
+	faucetBalance -= faucetAmount
+	calculatedWallet1Balance := wallet1Balance + faucetAmount
+	env.AssertFaucetBalance(faucetBalance)
+	env.TestEnv.AssertLedgerBalance(env.FaucetWallet, faucetBalance)
+	env.TestEnv.AssertLedgerBalance(env.Wallet1, calculatedWallet1Balance)
+
+	message := env.TestEnv.NewMessageBuilder().
+		LatestMilestonesAsParents().
+		FromWallet(env.GenesisWallet).
+		ToWallet(env.FaucetWallet).
+		Amount(faucetAmount).
+		Build().
+		Store().
+		BookOnWallets()
+
+	// Confirming milestone at message
+	_, _ = env.IssueMilestone(hornet.MessageIDs{message.StoredMessageID()}...)
```

Comment:
why are you wrapping a messageId in an array and then passing it as variadic? just pass `message.StoredMessageID()`